extensions/micrometer/deployment/src/test/java/io/quarkus/micrometer/deployment/binder/NettyMetricsTest.java

```diff
@@ -0,0 +1,229 @@
+        Assertions.assertNotNull(direct0);
+
+        RestAssured.get("/hello/Netty").then().body(Matchers.equalTo("hello Netty"));
+
+        Double heap1 = getValue(meters, tags(tags, HEAP_MEMORY));
+        Double direct1 = getValue(meters, tags(tags, DIRECT_MEMORY));
+
+        Assertions.assertTrue(heap0 <= heap1);
+        Assertions.assertTrue(direct0 <= direct1);
+    }
+
+    @Test
+    public void testNettyAllocatorMetrics() {
+        testNettyMetrics(5L, NettyAllocatorMetrics.class);
+    }
+
+    @Test
+    public void testNettyEventExecutorMetrics() {
+        testNettyMetrics(2L, NettyEventExecutorMetrics.class);
+    }
+
+    @Test
+    public void testAllocatorMetricsValues() {
+        testAllocatorMetricsValues(NAM_PBBA_TAGS);
+        testAllocatorMetricsValues(NAM_UNPBBA_TAGS);
+        testAllocatorMetricsValues(VX_NAM_PBBA_TAGS);
+        testAllocatorMetricsValues(VX_NAM_UNPBBA_TAGS);
+    }
+
+    @Test
+    public void testEventExecutorMetricsValues() throws Exception {
+        VertxInternal vi = (VertxInternal) vertx;
+        assertEventGroup(vi.getEventLoopGroup());
+        assertEventGroup(vi.getAcceptorEventLoopGroup());
+    }
+
+    private void assertEventGroup(EventLoopGroup group) throws Exception {
+        int tasks = 0;
+        for (EventExecutor ee : group) {
+            tasks++;
+        }
+        final CyclicBarrier allPendingTasksAreIn = new CyclicBarrier(tasks + 1);
+        CountDownLatch waitCollectingMeasures = new CountDownLatch(1);
+        List<Future<Future<?>>> pendingTasksCompleted = new ArrayList<>(tasks);
+        for (EventExecutor eventLoop : group) {
+            pendingTasksCompleted.add(eventLoop.submit(() -> {
+                try {
+                    Future<?> pendingTask = eventLoop.submit(() -> {
+                    });
+                    // this executor will have 1 pending task because is still running this current one
+                    allPendingTasksAreIn.await();
+                    waitCollectingMeasures.await();
+                    return pendingTask;
+                } catch (Throwable ignore) {
+                    return null;
+                }
+            }));
+        }
+        allPendingTasksAreIn.await();
+        List<Meter> meters = registry.getMeters();
+        // this would return 1 for everyone
+        for (EventExecutor eventLoop : group) {
+            checkMetrics(meters, eventLoop, 1);
+        }
+        waitCollectingMeasures.countDown();
+        for (Future<Future<?>> pendingTaskCompleted : pendingTasksCompleted) {
+            pendingTaskCompleted.get().get();
+        }
+        // this would return 0 for everyone
+        for (EventExecutor eventLoop : group) {
+            checkMetrics(meters, eventLoop, 0);
+        }
+    }
+
+    private void checkMetrics(List<Meter> meters, EventExecutor executor, int expected) {
+        if (executor instanceof SingleThreadEventExecutor) {
```

Comment:
I would make it fail if it won't match or we risk that a change in the class type would break silently

Reply:
Well, metrics are only added for this type ... so I guess we should only check for metrics for this type?

```
    public NettyEventExecutorMetrics(Iterable<EventExecutor> eventExecutors) {
        this.eventExecutors = eventExecutors;
    }

    @Override
    public void bindTo(MeterRegistry registry) {
        this.eventExecutors.forEach(eventExecutor -> {
            if (eventExecutor instanceof SingleThreadEventExecutor) {
```